### PR TITLE
Automatic test marking

### PR DESCRIPTION
**Pull request by jspaaks**
**List of related issues or pull requests**

Refs: -

**Describe the changes made in this pull request**

This PR adds automatic test annotation based on the file path and class name. The following markers have been added to`setup.cfg`:

1. ~~`bitbucket`: Bitbucket tests~~
1.  `checker`: Checker tests
1.  `compliance`: Compliance tests
1.  `github`: GitHub tests
1.  `gitlab`: GitLab tests
1.  `repo`: Repo tests

With these annotations, you can now more easily run a subset of tests, e.g. 

```shell
$ pytest -m 'repo'
========================================== test session starts ===========================================
platform linux -- Python 3.6.9, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /home/daisycutter/github/fair-software/howfairis, configfile: setup.cfg, testpaths: tests
plugins: requests-mock-1.8.0, cov-2.11.1
collected 59 items / 5 deselected / 54 selected                                                          

tests/github/fair_software/badge/test_repo_no_args.py .........                                    [ 16%]
tests/github/fair_software/badge/test_repo_with_branch_develop.py .........                        [ 33%]
tests/github/fair_software/badge/test_repo_with_branch_sha.py .........                            [ 50%]
tests/github/fair_software/badge/test_repo_with_branch_tag.py .........                            [ 66%]
tests/github/fair_software/badge/test_repo_with_path.py .........                                  [ 83%]
tests/gitlab/jspaaks/badge_test/test_repo_no_args.py .........                                     [100%]

==================================== 54 passed, 5 deselected in 0.28s ====================================
```

Markers can also be combined as follows:

```
$ pytest -m 'repo and not github'
========================================== test session starts ===========================================
platform linux -- Python 3.6.9, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /home/daisycutter/github/fair-software/howfairis, configfile: setup.cfg, testpaths: tests
plugins: requests-mock-1.8.0, cov-2.11.1
collected 59 items / 50 deselected / 9 selected                                                          

tests/gitlab/jspaaks/badge_test/test_repo_no_args.py .........                                     [100%]

==================================== 9 passed, 50 deselected in 0.18s ====================================
```

**Instructions to review the pull request**

```
pytest --markers
pytest -m 'repo'
pytest -m 'repo and not github'
```

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import pytest
+
+
+def pytest_collection_modifyitems(items):
+    for item in items:
+        if "bitbucket" in item.nodeid:
+            item.add_marker(pytest.mark.bitbucket)
+        if "heptapod" in item.nodeid:
+            item.add_marker(pytest.mark.heptapod)
```
Comment (fdiblen) [inline tests/conftest.py:9]:
```suggestion
```

File: setup.cfg [hunk 2]
```diff
@@ -25,6 +25,13 @@ source = howfairis
 
 [tool:pytest]
 testpaths = tests
+markers =
+  bitbucket: Bitbucket tests
```
Comment (fdiblen) [inline setup.cfg:29]:
```suggestion
```

**Review by fdiblen — CHANGES_REQUESTED** [review]
We should include bitbucket and heptapod related marks in a different PR.

**Comment by jspaaks** [comment]
suggestions committed

**Comment by sverhoeven** [comment]
This seems not that useful to me.

**Comment by jspaaks** [comment]
We decided this work isnt really needed at the moment, closing the PR

**Pull request closed without merging**